Common/EventId.h

```diff
@@ -0,0 +1,177 @@
+  RunLumi(RunType runNumber, LumiType lumiBlock) : run(runNumber), luminosityBlock(lumiBlock) {}
+
+  bool operator ==(const RunLumi& other) const
+  {
+    return run == other.run && luminosityBlock == other.luminosityBlock;
+  }
+
+  bool operator !=(const RunLumi& other) const
+  {
+    return run != other.run || luminosityBlock != other.luminosityBlock;
+  }
+
+  bool operator <(const RunLumi& other) const
+  {
+    if(run != other.run) return run < other.run;
+    return luminosityBlock < other.luminosityBlock;
+  }
+
+  std::string ToString() const
+  {
+    std::ostringstream ss;
+    ss << run << separator << luminosityBlock;
+    return ss.str();
+  }
+};
+
+std::ostream& operator<<(std::ostream& s, const RunLumi& runLumi)
+{
+  s << runLumi.ToString();
+  return s;
+}
+
+std::istream& operator>>(std::istream& s, RunLumi& runLumi)
+{
+  std::string str;
+  s >> str;
+  runLumi = RunLumi::FromString(str);
+  return s;
+}
+
+struct RunLumiEvent {
+  using RunType = RunLumi::RunType;
+  using LumiType = RunLumi::LumiType;
+  using EventType = unsigned long long;
+
+  static constexpr RunType undefinedRun = RunLumi::undefinedRun;
+  static constexpr LumiType undefinedLumi = RunLumi::undefinedLumi;
+  static constexpr EventType undefinedEvent = std::numeric_limits<EventType>::max();
+
+  static constexpr char separator = RunLumi::separator;
+
+  static const RunLumiEvent& Undefined()
+  {
+    static const RunLumiEvent undefined;
+    return undefined;
+  }
+
+  static RunLumiEvent FromString(const std::string& idStr)
+  {
+    const auto idStrings = Split(idStr);
+    if(idStrings.size() != 3)
+      throw std::invalid_argument("RunLumiEvent: invalid run:lumi string = '" + idStr + "'.");
+    return FromStrings(idStrings[0], idStrings[1], idStrings[2]);
+  }
+
+  static RunLumiEvent FromStrings(const std::string& runStr, const std::string& lumiStr, const std::string& eventStr)
+  {
+    const auto runLumi = RunLumi::FromStrings(runStr, lumiStr);
+    const auto event = Parse<EventType>(eventStr, "event");
+    return RunLumi(run, luminosityBlock, event)
+  }
+
+  RunType run{undefinedRun};
+  LumiType luminosityBlock{undefinedLumi};
+  EventType event{undefinedEvent};
+
+  RunLumiEvent() {}
+  RunLumiEvent(RunType runNumber, LumiType lumiBlock, EventType eventNumber)
+    : run(runNumber), luminosityBlock(lumiBlock), event(eventNumber) {}
+
+  bool operator ==(const RunLumiEvent& other) const
+  {
+    return run == other.run && luminosityBlock == other.luminosityBlock && event == other.event;
+  }
+
+  bool operator !=(const RunLumiEvent& other) const
+  {
+    return run != other.run || luminosityBlock != other.luminosityBlock || event != other.event;
+  }
+
+  bool operator <(const RunLumiEvent& other) const
+  {
+    if(run != other.run) return run < other.run;
+    if(luminosityBlock != other.luminosityBlock) return luminosityBlock < other.luminosityBlock;
+    return event < other.event;
+  }
+
+  std::string ToString() const
```

Comment:
(just for my knowledge). This function could be generalised and put in the class CorrectionBase or also in analysis tools. Is it a good idea?